src/jdk.incubator.code/share/classes/jdk/incubator/code/internal/ReflectMethods.java

```diff
@@ -2328,7 +2329,7 @@ public void visitUnary(JCTree.JCUnary tree) {
                 case POSTINC, POSTDEC, PREINC, PREDEC -> {
                     // Capture applying rhs and operation
                     Function<Value, Value> scanRhs = (lhs) -> {
-                        Value one = append(numericOneValue(tree.type));
+                        Value one = convert(append(numericOneValue(tree.type)), types.unboxedType(tree.type));
```

Comment:
I don't get the call to `types.uboxedType` -- `tree.type` is passed to `numericOneValue` which seems to assume that the type is a primitive type, right?

Reply:
Ah ok - I see:
```
                case CLASS -> numericOneValue(types.unboxedType(t));
```

So, `numericOneValue` also does unboxing. I think this code would be clearer by doing the unboxing first -- and then pass the unboxed type to both `numericOneValue` and `convert` (and then, we could remove the unboxing path from `numericOneValue`.

Reply:
> and make that consistent with `defaultValue`).

Strike that -- `defaultValue` also allows `nulls`. Anyway -- `numericOneValue` should only work on primitives, I think.

Reply:
For `defaultValue`, the passed type can be primitive or reference. For `numericOneValue` we can only accept a primitive type.